### ISSUE TITLE
Xgboost 2.0.0  · dmlc/xgboost

### ISSUE THREAD

**Issue by ShellLM** [irthomasthomas/undecidability#878]
- [ ] [Release Release 2.0.0 stable · dmlc/xgboost](https://github.com/dmlc/xgboost/releases/tag/v2.0.0)

# Release Release 2.0.0 stable · dmlc/xgboost

## Snippet

We are excited to announce the release of XGBoost 2.0. This note will begin by covering some overall changes and then highlight specific updates to the package.

## Initial work on multi-target trees with vector-leaf outputs

We have been working on vector-leaf tree models for multi-target regression, multi-label classification, and multi-class classification in version 2.0. Previously, XGBoost would build a separate model for each target. However, with this new feature that's still being developed, XGBoost can build one tree for all targets. The feature has multiple benefits and trade-offs compared to the existing approach. It can help prevent overfitting, produce smaller models, and build trees that consider the correlation between targets. In addition, users can combine vector leaf and scalar leaf trees during a training session using a callback. Please note that the feature is still a working in progress, and many parts are not yet available. See [#9043](https://github.com/dmlc/xgboost/issues/9043) for the current status. Related PRs: ([#8538](https://github.com/dmlc/xgboost/pull/8538), [#8697](https://github.com/dmlc/xgboost/pull/8697), [#8902](https://github.com/dmlc/xgboost/pull/8902), [#8884](https://github.com/dmlc/xgboost/pull/8884), [#8895](https://github.com/dmlc/xgboost/pull/8895), [#8898](https://github.com/dmlc/xgboost/pull/8898), [#8612](https://github.com/dmlc/xgboost/pull/8612), [#8652](https://github.com/dmlc/xgboost/pull/8652), [#8698](https://github.com/dmlc/xgboost/pull/8698), [#8908](https://github.com/dmlc/xgboost/pull/8908), [#8928](https://github.com/dmlc/xgboost/pull/8928), [#8968](https://github.com/dmlc/xgboost/pull/8968), [#8616](https://github.com/dmlc/xgboost/pull/8616), [#8922](https://github.com/dmlc/xgboost/pull/8922), [#8890](https://github.com/dmlc/xgboost/pull/8890), [#8872](https://github.com/dmlc/xgboost/pull/8872), [#8889](https://github.com/dmlc/xgboost/pull/8889), [#9509](https://github.com/dmlc/xgboost/pull/9509)) Please note that, only the hist (default) tree method on CPU can be used for building vector leaf trees at the moment.

## New device parameter

A new device parameter is set to replace the existing gpu_id, gpu_hist, gpu_predictor, cpu_predictor, gpu_coord_descent, and the PySpark specific parameter use_gpu. Onward, users need only the device parameter to select which device to run along with the ordinal of the device. For more information, please see our document page (https://xgboost.readthedocs.io/en/stable/parameter.html#general-parameters) . For example, with device="cuda", tree_method="hist", XGBoost will run the hist tree method on GPU. ([#9363](https://github.com/dmlc/xgboost/pull/9363), [#8528](https://github.com/dmlc/xgboost/pull/8528), [#8604](https://github.com/dmlc/xgboost/pull/8604), [#9354](https://github.com/dmlc/xgboost/pull/9354), [#9274](https://github.com/dmlc/xgboost/pull/9274), [#9243](https://github.com/dmlc/xgboost/pull/9243), [#8896](https://github.com/dmlc/xgboost/pull/8896), [#9129](https://github.com/dmlc/xgboost/pull/9129), [#9362](https://github.com/dmlc/xgboost/pull/9362), [#9402](https://github.com/dmlc/xgboost/pull/9402), [#9385](https://github.com/dmlc/xgboost/pull/9385), [#9398](https://github.com/dmlc/xgboost/pull/9398), [#9390](https://github.com/dmlc/xgboost/pull/9390), [#9386](https://github.com/dmlc/xgboost/pull/9386), [#9412](https://github.com/dmlc/xgboost/pull/9412), [#9507](https://github.com/dmlc/xgboost/pull/9507), [#9536](https://github.com/dmlc/xgboost/pull/9536)). The old behavior of gpu_hist is preserved but deprecated. In addition, the predictor parameter is removed.

## hist is now the default tree method

Starting from 2.0, the hist tree method will be the default. In previous versions, XGBoost chooses approx or exact depending on the input data and training environment. The new default can help XGBoost train models more efficiently and consistently. ([#9320](https://github.com/dmlc/xgboost/pull/9320), [#9353](https://github.com/dmlc/xgboost/pull/9353))

## GPU-based approx tree method

There's initial support for using the approx tree method on GPU. The performance of the approx is not yet well optimized but is feature complete except for the JVM packages. It can be accessed through the use of the parameter combination device="cuda", tree_method="approx". ([#9414](https://github.com/dmlc/xgboost/pull/9414), [#9399](https://github.com/dmlc/xgboost/pull/9399), [#9478](https://github.com/dmlc/xgboost/pull/9478)). Please note that the Scala-based Spark interface is not yet supported.

## Optimize and bound the size of the histogram on CPU, to control memory footprint

XGBoost has a new parameter max_cached_hist_node for users to limit the CPU cache size for histograms. It can help prevent XGBoost from caching histograms too aggressively. Without the cache, performance is likely to decrease. However, the size of the cache grows exponentially with the depth of the tree. The limit can be crucial when growing deep trees. In most cases, users need not configure this parameter as it does not affect the model's accuracy. ([#9455](https://github.com/dmlc/xgboost/pull/9455), [#9441](https://github.com/dmlc/xgboost/pull/9441), [#9440](https://github.com/dmlc/xgboost/pull/9440), [#9427](https://github.com/dmlc/xgboost/pull/9427), [#9400](https://github.com/dmlc/xgboost/pull/9400)).

Along with the cache limit, XGBoost also reduces the memory usage of the hist and approx tree method on distributed systems by cutting the size of the cache by half. ([#9433](https://github.com/dmlc/xgboost/pull/9433))

## Improved external memory support

There is some exciting development around external memory support in XGBoost. It's still an experimental feature, but the performance has been significantly improved with the default hist tree method. We replaced the old file IO logic with memory map. In addition to performance, we have reduced CPU memory usage and added extensive documentation. Beginning from 2.0.0, we encourage users to try it with the hist tree method when the memory saving by QuantileDMatrix is not sufficient. ([#9361](https://github.com/dmlc/xgboost/pull/9361), [#9317](https://github.com/dmlc/xgboost/pull/9317), [#9282](https://github.com/dmlc/xgboost/pull/9282), [#9315](https://github.com/dmlc/xgboost/pull/9315), [#8457](https://github.com/dmlc/xgboost/pull/8457))

## Learning to rank

We created a brand-new implementation for the learning-to-rank task. With the latest version, XGBoost gained a set of new features for ranking task including:

- A new parameter lambdarank_pair_method for choosing the pair construction strategy.
- A new parameter lambdarank_num_pair_per_sample for controlling the number of samples for each group.
- An experimental implementation of unbiased learning-to-rank, which can be accessed using the lambdarank_unbiased parameter.
- Support for custom gain function with NDCG using the ndcg_exp_gain parameter.
- Deterministic GPU computation for all objectives and metrics.
- NDCG is now the default objective function.
- Improved performance of metrics using caches.
- Support scikit-learn utilities for XGBRanker.
- Extensive documentation on how learning-to-rank works with XGBoost.

For more information, please see the tutorial. Related PRs: ([#8771](https://github.com/dmlc/xgboost/pull/8771), [#8692](https://github.com/dmlc/xgboost/pull/8692), [#8783](https://github.com/dmlc/xgboost/pull/8783), [#8789](https://github.com/dmlc/xgboost/pull/8789), [#8790](https://github.com/dmlc/xgboost/pull/8790), [#8859](https://github.com/dmlc/xgboost/pull/8859), [#8887](https://github.com/dmlc/xgboost/pull/8887), [#8893](https://github.com/dmlc/xgboost/pull/8893), [#8906](https://github.com/dmlc/xgboost/pull/8906), [#8931](https://github.com/dmlc/xgboost/pull/8931), [#9075](https://github.com/dmlc/xgboost/pull/9075), [#9015](https://github.com/dmlc/xgboost/pull/9015), [#9381](https://github.com/dmlc/xgboost/pull/9381), [#9336](https://github.com/dmlc/xgboost/pull/9336), [#8822](https://github.com/dmlc/xgboost/pull/8822), [#9222](https://github.com/dmlc/xgboost/pull/9222), [#8984](https://github.com/dmlc/xgboost/pull/8984), [#8785](https://github.com/dmlc/xgboost/pull/8785), [#8786](https://github.com/dmlc/xgboost/pull/8786), [#8768](https://github.com/dmlc/xgboost/pull/8768))

## Automatically estimated intercept

In the previous version, base_score was a constant that could be set as a training parameter. In the new version, XGBoost can automatically estimate this parameter based on input labels for optimal accuracy. ([#8539](https://github.com/dmlc/xgboost/pull/8539), [#8498](https://github.com/dmlc/xgboost/pull/8498), [#8272](https://github.com/dmlc/xgboost/pull/8272), [#8793](https://github.com/dmlc/xgboost/pull/8793), [#8607](https://github.com/dmlc/xgboost/pull/8607))

## Quantile regression

The XGBoost algorithm now supports quantile regression, which involves minimizing the quantile loss (also called "pinball loss"). Furthermore, XGBoost allows for training with multiple target quantiles simultaneously with one tree per quantile. ([#8775](https://github.com/dmlc/xgboost/pull/8775), [#8761](https://github.com/dmlc/xgboost/pull/8761), [#8760](https://github.com/dmlc/xgboost/pull/8760), [#8758](https://github.com/dmlc/xgboost/pull/8758), [#8750](https://github.com/dmlc/xgboost/pull/8750))

## L1 and Quantile regression now supports learning rate

Both objectives use adaptive trees due to the lack of proper Hessian values. In the new version, XGBoost can scale the leaf value with the learning rate accordingly. ([#8866](https://github.com/dmlc/xgboost/pull/8866))

## Export cut value

Using the Python or the C package, users can export the quantile values (not to be confused with quantile regression) used for the hist tree method. ([#9356](https://github.com/dmlc/xgboost/pull/9356))

## column-based split and federated learning

We made progress on column-based split for federated learning. In 2.0, both approx, hist, and hist with vector leaf can work with column-based data split, along with support for vertical federated learning. Work on GPU support is still on-going, stay tuned. ([#8576](https://github.com/dmlc/xgboost/pull/8576), [#8468](https://github.com/dmlc/xgboost/pull/8468), [#8442](https://github.com/dmlc/xgboost/pull/8442), [#8847](https://github.com/dmlc/xgboost/pull/8847), [#8811](https://github.com/dmlc/xgboost/pull/8811), [#8985](https://github.com/dmlc/xgboost/pull/8985), [#8623](https://github.com/dmlc/xgboost/pull/8623), [#8568](https://github.com/dmlc/xgboost/pull/8568), [#8828](https://github.com/dmlc/xgboost/pull/8828), [#8932](https://github.com/dmlc/xgboost/pull/8932), [#9081](https://github.com/dmlc/xgboost/pull/9081), [#9102](https://github.com/dmlc/xgboost/pull/9102), [#9103](https://github.com/dmlc/xgboost/pull/9103), [#9124](https://github.com/dmlc/xgboost/pull/9124), [#9120](https://github.com/dmlc/xgboost/pull/9120), [#9367](https://github.com/dmlc/xgboost/pull/9367), [#9370](https://github.com/dmlc/xgboost/pull/9370), [#9343](https://github.com/dmlc/xgboost/pull/9343), [#9171](https://github.com/dmlc/xgboost/pull/9171), [#9346](https://github.com/dmlc/xgboost/pull/9346), [#9270](https://github.com/dmlc/xgboost/pull/9270), [#9244](https://github.com/dmlc/xgboost/pull/9244), [#8494](https://github.com/dmlc/xgboost/pull/8494), [#8434](https://github.com/dmlc/xgboost/pull/8434), [#8742](https://github.com/dmlc/xgboost/pull/8742), [#8804](https://github.com/dmlc/xgboost/pull/8804), [#8710](https://github.com

#### {'label-name': 'XGBoost', 'label-description': 'A popular machine learning library for gradient boosting frameworks.', 'gh-repo': 'dmlc/xgboost', 'confidence': 81.76}

**Comment by ShellLM** [comment]
### Related content
#743 similarity score: 0.86
#640 similarity score: 0.86
#625 similarity score: 0.86
#649 similarity score: 0.86
#304 similarity score: 0.86
#499 similarity score: 0.86
